python/tank/pipelineconfig.py

```diff
@@ -34,6 +34,8 @@ class PipelineConfiguration(object):
     create directly via constructor.
     """
 
+    _UNKNOWN_PROJECT_ID = "unknown_project_id"
```

Comment:
And I think a comment around this somewhere would be helpful. It's not totally obvious for a new reader of the code that we cannot use None as an "uninitialized marker" here because `None` actually has a meaning as a value.